### PR TITLE
Fix minor documentation and comment typos

### PR DESCRIPTION
**Pull request by aSemy**
Very minor points. I'm mostly committing so my IDE will show fewer warnings.

### CODE REVIEW

File: formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt [hunk 1]
```diff
@@ -268,7 +268,7 @@ internal open class StreamingJsonDecoder(
 
     override fun decodeBoolean(): Boolean {
         /*
-         * We prohibit non true/false boolean literals at all as it is considered way too error-prone,
+         * We prohibit non-true/false boolean literals at all as it is considered way too error-prone,
```
Comment (sandwwraith) [inline formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt:271]: Here it would actually change the meaning. The original comment "non true/false" means "any literal that is not equal to 'true' or 'false'", while changed versions read as "We prohibit non-true OR false boolean literals".

You can rephrase this to "We prohibit any boolean literal that is not strictly 'true' or 'false'"

**Review by sandwwraith — CHANGES_REQUESTED** [review]

File: formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt (hunk 1 above)
Comment (aSemy) [reply formats/json/commonMain/src/kotlinx/serialization/json/internal/StreamingJsonDecoder.kt]: good spot, thanks. Your rephrasing is much more clear.

**Review by sandwwraith — APPROVED** [review]
